aim/aim_manager.py

```diff
@@ -150,6 +151,21 @@ def find(self, context, resource_class, **kwargs):
                 self._make_resource(context.db_session, resource_class, obj))
         return result
 
+    def get_status(self, context, resource):
```

Comment:
Status is operational state while the resource represents the configuration/desired state. I think it makes sense to keep the two separate. Note, that, to get the status of a resource, you don't need to "retrieve" the resource from AimManager by calling get(). You can just create the Python object for a resource with the minimum required attributes and pass it to get_status().